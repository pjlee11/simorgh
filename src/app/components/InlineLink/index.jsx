import styled from 'styled-components';
import {
  C_BLUEJAY,
  C_BLUEJAY_LHT,
  C_OAT_LHT,
  C_PEBBLE,
} from '../../lib/constants/styles';

const InlineLink = styled.a`
  color: ${C_BLUEJAY};
  padding: 2px;
  border-bottom: 1px solid ${C_BLUEJAY};
  text-decoration: none;
  &:visited {
    color: ${C_PEBBLE};
    border-bottom: 1px solid ${C_PEBBLE};
  }
  &:focus {
    background-color: ${C_BLUEJAY};
    border-bottom: 1px solid ${C_OAT_LHT};
    color: ${C_OAT_LHT};
  }
  &:hover {
    color: ${C_BLUEJAY};
    background-color: ${C_BLUEJAY_LHT};
    border-bottom: 1px solid ${C_BLUEJAY};
  }
`;

export default InlineLink;
