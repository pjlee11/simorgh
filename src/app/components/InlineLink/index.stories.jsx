import React from 'react';
import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
import InlineLink from './index';

storiesOf('InlineLink', module).add('default', () => (
  <InlineLink href="https://www.bbc.com/news">BBC News</InlineLink>
));
