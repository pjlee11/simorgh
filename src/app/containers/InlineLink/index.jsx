import React from 'react';
import { Link } from 'react-router-dom';
import { string, node } from 'prop-types';
import InlineLink from '../../components/InlineLink';

const InternalInlineLink = InlineLink.withComponent(Link);

const InlineLinkContainer = ({ href, children, ...rest }) => {
  const regex = '^/news/articles/c[a-zA-Z0-9]{10}o$';

  // if URL matches a valid route, use a react-router link
  if (href.match(regex)) {
    return (
      <InternalInlineLink to={href} {...rest}>
        {children}
      </InternalInlineLink>
    );
  }

  // else return a normal hyperlink
  return (
    <InlineLink href={href} {...rest}>
      {children}
    </InlineLink>
  );
};

InlineLinkContainer.propTypes = {
  href: string.isRequired,
  children: node.isRequired,
};

export default InlineLinkContainer;
