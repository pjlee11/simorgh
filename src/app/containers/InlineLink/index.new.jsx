import React from 'react';
import { Link } from 'react-router-dom';
import pathToRegexp from 'path-to-regexp';
import InlineLink from '../../components/InlineLink';
import Blocks from '../Blocks';
import fragment from '../Fragment';
import { inlineLinkModelPropTypes } from '../../models/propTypes/inlineLink';
import { articleRegexPath } from '../../routes';

const InternalInlineLink = InlineLink.withComponent(Link);

const componentsToRender = { fragment };

const InlineLinkContainer = ({ locator, blocks }) => {
  const regexp = pathToRegexp(articleRegexPath, [], {
    start: false,
    end: false,
  });

  const result = regexp.exec(locator);
  // if URL matches a valid route, use a react-router link
  if (result) {
    // the path is the first item in the array
    const path = result[0];
    return (
      <InternalInlineLink to={path}>
        <Blocks blocks={blocks} componentsToRender={componentsToRender} />
      </InternalInlineLink>
    );
  }

  // else return a normal hyperlink
  return (
    <InlineLink href={locator}>
      <Blocks blocks={blocks} componentsToRender={componentsToRender} />
    </InlineLink>
  );
};

InlineLinkContainer.propTypes = inlineLinkModelPropTypes;

export default InlineLinkContainer;
