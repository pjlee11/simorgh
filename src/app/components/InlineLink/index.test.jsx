import React from 'react';
import InlineLink from './index';

import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';

describe(`InlineLink`, () => {
  shouldMatchSnapshot(
    'should render correctly',
    <InlineLink href="https://www.bbc.com/news">BBC News</InlineLink>,
  );
});
