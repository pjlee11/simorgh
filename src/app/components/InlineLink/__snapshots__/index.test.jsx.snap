// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLink should render correctly 1`] = `
.c0 {
  color: #0F556C;
  padding: 2px;
  border-bottom: 1px solid #0F556C;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:visited {
  color: #5C5752;
  border-bottom: 1px solid #5C5752;
}

.c0:focus {
  background-color: #0F556C;
  border-bottom: 1px solid #F5F3F1;
  color: #F5F3F1;
}

.c0:hover {
  color: #0F556C;
  background-color: #C3DEE7;
  border-bottom: 1px solid #0F556C;
}

<a
  className="c0"
  href="https://www.bbc.com/news"
>
  BBC News
</a>
`;
